### PR TITLE
[Merged by Bors] - build: update k8 crates

### PR DESCRIPTION
**Pull request by morenol**

### CODE REVIEW

File: crates/fluvio-cli/src/error.rs [hunk 1]
```diff
@@ -32,10 +32,6 @@ pub enum CliError {
     #[error("Kubernetes config error: {0}")]
     K8ConfigError(#[from] k8_config::ConfigError),
```
Comment (sehz) [inline crates/fluvio-cli/src/error.rs:33]: do we still need this one?

File: crates/fluvio-cli/src/profile/sync/k8.rs [hunk 2]
```diff
@@ -112,22 +112,22 @@ pub async fn set_k8_context(opt: K8Opt, external_addr: String) -> Result<Profile
...
         .retrieve_item::<ServiceSpec, _>(&InputObjectMeta::named("fluvio-sc-public", ns))
         .await
     {
         Ok(svc) => svc,
-        Err(err) => match err {
-            k8_client::ClientError::ApiResponse(status)
-                if status.code == Some(StatusCode::NOT_FOUND.as_u16()) =>
+        Err(err) => {
+            if let Some(MetaStatus {
+                code: Some(404), ..
```
Comment (sehz) [inline crates/fluvio-cli/src/profile/sync/k8.rs:123]: use http crate?

**Review by sehz — CHANGES_REQUESTED** [review]
Nice cleanup.  couple of nits

File: crates/fluvio-cli/src/error.rs (hunk 1 above)
Comment (morenol) [reply crates/fluvio-cli/src/error.rs:33]: I didn't update it, but it seems to compile successfully without this

File: crates/fluvio-cli/src/profile/sync/k8.rs (hunk 2 above)
Comment (morenol) [reply crates/fluvio-cli/src/profile/sync/k8.rs:123]: Not sure what you mean. 

This is just replacing the old

```
k8_client::ClientError::ApiResponse(status)
                if status.code == Some(StatusCode::NOT_FOUND.as_u16()) =>
```
                
using the anyhow downcasting to extract the status code from the inner error.

**Review by sehz — APPROVED** [review]
LGTM

**Pull request closed without merging**